src/VisualStudio/VisualBasic/Impl/LanguageService/VisualBasicPackage.vb

```diff
@@ -58,20 +58,16 @@ Namespace Microsoft.VisualStudio.LanguageServices.VisualBasic
             _comAggregate = Interop.ComAggregate.CreateAggregatedObject(Me)
         End Sub
 
-        Protected Overrides Function CreateWorkspace() As VisualStudioWorkspaceImpl
-            Return Me.ComponentModel.GetService(Of VisualStudioWorkspaceImpl)
-        End Function
-
         Protected Overrides Async Function InitializeAsync(cancellationToken As CancellationToken, progress As IProgress(Of ServiceProgressData)) As Task
             Try
                 Await MyBase.InitializeAsync(cancellationToken, progress).ConfigureAwait(True)
                 Await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken)
 
                 RegisterLanguageService(GetType(IVbCompilerService), Function() Task.FromResult(_comAggregate))
 
-                Dim workspace = Me.ComponentModel.GetService(Of VisualStudioWorkspaceImpl)()
                 RegisterService(Of IVbTempPECompilerFactory)(
                     Async Function(ct)
+                        Dim workspace = Me.ComponentModel.GetService(Of VisualStudioWorkspaceImpl)()
```

Comment:
In the C# version you get `VisualStudioWorkspace` instead. Not sure it matters.

Reply:
It doesn't matter from a functional perspective, but I've been on a long crusade to not have code ever mentioned VisualStudioWorkspaceImpl unless they actually depend on the implementation details. At least makes it easy to figure out what bits of our code are doing evil versus just using a public API.